View Controllers/TimelineViewController.m

```diff
@@ -149,18 +149,19 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
 
 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
-//    UINavigationController *navigationController = [segue destinationViewController];
-//    ComposeViewController *composeController = (ComposeViewController*)navigationController.topViewController;
-//    composeController.delegate = self;
-    
-    UITableViewCell *tappedCell = sender;
-    NSIndexPath *indexPath = [self.tableView indexPathForCell:tappedCell];
-    Tweet *tweet = self.arrayOfTweets[indexPath.row];
-
-    TweetDetailsViewController *tweetViewController = [segue destinationViewController];
-    tweetViewController.tweet = tweet;
-    NSLog(@"Tapping on a tweet");
-    
+    if([sender isKindOfClass:[UIBarButtonItem class]]) {
```

Comment:
This gives me pause. I would think using a segue identifier would make more sense here. Let's discuss why this approach and what could possibly go wrong with it.